pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/QueryRunner.java

```diff
@@ -112,13 +120,14 @@ public void processQuery(DistributedStagePlan distributedStagePlan, ExecutorServ
       // TODO: make server query request return via mailbox, this is a hack to gather the non-streaming data table
       // and package it here for return. But we should really use a MailboxSendOperator directly put into the
       // server executor.
-      List<ServerQueryRequest> serverQueryRequests =
-          ServerRequestUtils.constructServerQueryRequest(distributedStagePlan, requestMetadataMap,
-              _helixPropertyStore);
+      List<ServerPlanRequestContext> serverQueryRequests = constructServerQueryRequests(distributedStagePlan,
+          requestMetadataMap, _helixPropertyStore);
```

Comment:
The goal here is for the future 

1. by returning the `ServerPlanRequestContext`, we will be able to process the compiled pinotQuery using v1 engine; but also potentially process other part of the plan like dynamic filtering or anything that depend on another stage in leaf node
2. able to later directly generate the operator chain instead of having to run another planMaker step.
3.